_posts/2025-03-31-mmaler-blogpost-1-intro.adoc

```diff
@@ -0,0 +1,133 @@
+
+== Even better Quarkus? Here’s what new tools are on the way
+
+=== Panache 2: Simplifying data access
+
+Hibernate ORM with Panache is a lightweight abstraction layer built on top of Hibernate ORM that simplifies data interactions and reduces boilerplate code.
+With the introduction of Panache 2, Quarkus expands its capabilities by integrating Hibernate Data Repositories and Jakarta Data under a unified API.
+This unified approach makes adopting, switching between, or combining different data access paradigms easier.
+By leveraging Hibernate's robust capabilities, Panache 2 enables developers to write concise and expressive queries while maintaining full compatibility with standard JPA features.
+This results in cleaner, more maintainable applications and an intuitive approach to database management.
+
+Panache 2 also introduces several key enhancements:
+Supports mixing stateless, managed, blocking, and reactive operations in a single API.
+Allows for the definition of generic or specialized entities (stateless, managed, blocking, reactive) and enables them to switch to another mode of operation via an API.
+
+* Unifies entity operations and repository operations by using nested interfaces.
+* Solves the user experience issue of repository injection by obtaining repository instances via generated static methods on the metamodel.
+* Supports type-unsafe repositories that are out of the box for every entity and every mode of operation (stateless, managed, blocking, reactive).
+* Provides type-safe repositories in addition to, or mixed with, type-unsafe repositories.
+
+These improvements make Panache 2 a powerful tool for developers looking for flexibility, efficiency, and ease of use when working with data in Quarkus.
+
+Of course, there's more exciting stuff on the horizon, but this is a blog post, not release notes! ;)
+
+== “OK, I would like to try it, but is it easy enough to migrate my workflow to Quarkus?”
+
+Migrating to a new framework can feel daunting, even when it promises better performance, lower costs, and an improved developer experience.
+It’s like being offered a better house in a better neighborhood, but hesitating because of the hassle of packing, moving, and settling in.
+
+With Quarkus, the transition doesn’t have to be painful.
+Thanks to its compatibility with existing Java standards, support for Jakarta EE and Spring APIs, and an extensive extension ecosystem, many projects can migrate incrementally, without rewriting everything from scratch.
+For example, the Spring compatibility layer allows reuse of familiar annotations and patterns, see the link:https://quarkus.io/guides/spring-di[Spring DI guide] to learn more.
+
+And if you need help? You’re not alone.
+Quarkus experts, like Roberto Cortez, can visit your company, demo what Quarkus can do, and guide you through the entire migration process—from architecture review to production readiness.
```

Comment:
@radcortez you wanna be called out to come and visit ? :)

Is this really something we want to push on quarkus.io community site ?

Reply:
Many on the Quarkus team also do that work. I can see that it can pass as being the work of a single person when it is not. I think it would be better to remove it.

Reply:
Let me patch it :)